src/cmd/singularity/cli/help.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cli
+
+import (
+	"os"
+
+	"github.com/singularityware/singularity/src/docs"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	HelpCmd.Flags().SetInterspersed(false)
+	SingularityCmd.SetHelpCommand(HelpCmd)
+
+	SingularityCmd.AddCommand(HelpCmd)
+}
+
+// HelpCmd singularity help
+var HelpCmd = &cobra.Command{
+	DisableFlagsInUseLine: true,
+	PreRun:                sylabsToken,
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) == 0 {
+			cmd.Root().Help()
+			return
+		}
+
+		c, _, e := cmd.Root().Find(args)
+		if _, err := os.Stat(args[0]); err == nil {
+			// Help prints (if set) the sourced %help section on the definition file
+			a := []string{"/bin/cat", "/.singularity.d/runscript.help"}
+			execStarter(cmd, args[0], a, "")
```

Comment:
We should not be using `execStarter` here. We just need to set up a bare `EngineConfig`, put it into `CommonConfig`, and use `exec.Pipe`